source/MaterialXView/Handlers/GeometryHandler.h

```diff
@@ -31,45 +37,61 @@ class GeometryHandler
         return _identifier;
     }
 
+    /// Input properties that can be used when computing buffer data
+    struct InputProperties
+    {
+        /// Output screen width
+        unsigned int screenWidth;
+        /// Output screen height
+        unsigned int screenHeight;
+        /// Output screen offset
+        unsigned int screenOffset;
+
+        /// Program input constructor
+        InputProperties(unsigned int sw=512, unsigned sh=512, unsigned int so=20)
+            : screenWidth(sw)
+            , screenHeight(sh)
+            , screenOffset(so)
+        {}
+    };
+
     /// Create indexing data for geometry. The indexing is assumed to 
     /// be for a set of triangles. That is every 3 values index triangle data.
     /// @param bufferSize The size of the returned buffer in bytes.
-    /// @param indexCount Number of indices in the indexing
-    virtual unsigned int* getIndexing(size_t &bufferSize, unsigned int& indexCount) = 0;
+    virtual GeometryIndexBuffer& getIndexing(size_t &bufferSize) = 0;
 
     /// Create position data. Each element is assumed to be 3 float in size.
     /// @param bufferSize The size of the returned buffer in bytes.
-    /// @param screenWidth Width of screen. Required to initialzie screen positions. Defaults to 0.
-    /// @param screenHeight Height of screen. Required to initialzie screen positions. Defaults to 0.
-    virtual float* getPositions(size_t &bufferSize, unsigned screenWidth=0, unsigned int screenHeight=0) = 0;
+    /// @param properties Input properties.
+    virtual FloatGeometryBuffer& getPositions(size_t &bufferSize, const InputProperties& properties) = 0;
```

Comment:
Would it be better to give the InputProperties in the constructor, and save a reference to it? To avoid making the interface for position data different from all other data.